### PR TITLE
feat(start android): extend the --detach flag to wait for appium/android

### PR DESCRIPTION
**Pull request by sjelin**
The next installment of my long running "make it easier to everything in a script" series

### CODE REVIEW

File: lib/cmds/opts.ts [hunk 1]
```diff
@@ -42,9 +42,11 @@ var opts: Options = {};
 opts[OUT_DIR] = new Option(OUT_DIR, 'Location to output/expect', 'string', Config.getSeleniumDir());
 opts[SELENIUM_PORT] =
     new Option(SELENIUM_PORT, 'Optional port for the selenium standalone server', 'string', '4444');
-opts[APPIUM_PORT] = new Option(APPIUM_PORT, 'Optional port for the appium server', 'string');
+opts[APPIUM_PORT] =
+    new Option(APPIUM_PORT, 'Optional port for the appium server', 'string', '4723');
```
Comment (juliemr) [inline lib/cmds/opts.ts]: Why is this one a number and the next one a string?

File: lib/cmds/start.ts [hunk 2]
```diff
@@ -280,38 +288,112 @@ function killAppium() {
...
-  })(100);
+      child.on('exit', (code: number, signal: string) => {
+        if (!done) {
+          done = true;
+          resolve();
+        }
+      });
+      setTimeout(() => {
+        if (!done) {
+          done = false;
```
Comment (juliemr) [inline lib/cmds/start.ts]: Should be `done = true`?

Comment (sjelin) [inline lib/cmds/start.ts]: Yup :sweat:

File: lib/cmds/opts.ts (hunk 1 above)
Comment (sjelin) [inline lib/cmds/opts.ts]: In general, ports should be strings.  But arithmetic is done on the android port number, and it has to be between 5554 and 5586 anyway, so I decided to make it a number in this case.

**Pull request merged**